### DISCUSSION TITLE
Set attribute - how to use?

### DISCUSSION THREAD

**Discussion by peace2000** [uBlockO/uBO-Scriptlets#18] [Q&A] [answered]
Hi, how can I use set attribute scriplet in this case?

I want to add `[selected="SELECTED"]` as a part of `option[value="Solved (Permanently)"]`

![image](https://user-images.githubusercontent.com/17256841/159491498-7296885e-2ff7-445a-974b-e517258b41e7.png)

like this:

![image](https://user-images.githubusercontent.com/17256841/159492547-81b0e630-de2e-42c5-a2ac-415364dd398e.png)

There is a sample how to use sa.js https://raw.githubusercontent.com/uBlock-user/uBO-Scriptlets/master/scriptlets.js but I don't understand it properly.

**Comment by uBlock-user** [comment, marked as answer]
> example.com##+js(sa, preload, none, video)

first param is name of attribute
second param is attribute value
third param is CSS selector

**Comment by peace2000** [comment]
Are procedural CSS selector values unsupported?

**Reply by uBlock-user** [reply]
No and no for any other such scriptlets either.

**Reply by krystian3w** [reply]
Fine support CSS4 `:has(` in Chrome 105+.

**Reply by uBlock-user** [reply]
Post example which can be queried to `document.querySelectorAll.`

**Comment by peace2000** [comment]
Ok, thanks.

**Reply by uBlock-user** [reply]
Why do you need procedural selection support ?

**Reply by peace2000** [reply]
I'm using this script on a private site so I cannot give a link to exactly show my case but I could target this script better in my case if I could use a procedural css `:has()` with this script.

**Reply by uBlock-user** [reply]
All CSS selectors are fed to `document.querySelectorAll` so not possible.

Once [:has()](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) is adopted, this won't be an issue.

**Comment by peace2000** [comment]
All right, I'll have to wait for that then.

Other question... How can I set a value that contains a comma?

`example.com##+js(sa, attribute, 7,5, css)`

Does not work.

I tried `"7,5"` and `[7,5]` but they don't work either.

**Reply by uBlock-user** [reply]
> example.com##+js(sa, attribute, 7,5, css)

`example.com##+js(sa, attribute, 7\,5, css)`

**Comment by peace2000** [comment]
Hi, thanks but `7\,5` sets also an extra space between the comma and the five.

![image](https://user-images.githubusercontent.com/17256841/160073291-ceb1231e-c3a7-4ad3-8efe-bf0bf0b380eb.png)

Any way to get `7,5` without an extra space?

**Reply by uBlock-user** [reply]
I don't know how, probably something to address in uBO.

**Comment by peace2000** [comment]
Another question.. Can I use `stay` switch with `set attribute`? (I refer to the same switch that can be paired with uBO's `remove attribute` scriplet).

**Reply by uBlock-user** [reply]
`stay` is `loop` here.

**Reply by peace2000** [reply]
Works, thanks! Btw, shouldn't that information be listed here in example section? https://ubo-scriptlets.pages.dev/scriptlets.js

**Reply by uBlock-user** [reply]
It's optional.
